bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/AddonInfoRegistry.java

```diff
@@ -40,38 +43,94 @@ public class AddonInfoRegistry {
     private final Collection<AddonInfoProvider> addonInfoProviders = new CopyOnWriteArrayList<>();
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
-    protected void addAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
+    public void addAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
```

Comment:
Afaics, you made this public only for the tests to get access.
Could you instead move the test into the same package? Then it would have access to the protected method.

Reply:
> Could you instead move the test into the same package? 

Umm. If I would do that, then surely it would mean that all the org.junit and org.mockito dependencies in the test module, would need to be pulled into the runtime module as well?

Reply:
Err, no. You can move it in the same package, while keeping it in the `src/test/java` source folder, so that the test class won't be included into the runtime bundle.

Reply:
Ok, I will do that.